### PR TITLE
[WIP] Add jest frontend tests

### PR DESCRIPTION
**Pull request by woosie10**
Add at least a snapshot test for each component

### CODE REVIEW

**Comment by ChrisTerBeke** [comment]
Nice :)

**Comment by woosie10** [comment]
I messed up my merge as I didn't realise you'd (Chris) had already done it, but it should be fine now.

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# top-most EditorConfig file
+root = true
+
+[*]
+indent_style = space
+indent_size = 2
```
Comment (ChrisTerBeke) [inline .editorconfig]: Aren't we using 4?

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -10,7 +10,7 @@ before_script:
 test:
   stage: test
   script:
-    # TODO: run actual tests here when we have them, but for now we just build storybook to test
+    - yarn test
```
Comment (ChrisTerBeke) [inline .gitlab-ci.yml]: We should run `yarn test:coverage` instead, and also add the minimum coverage percentages to package.json (see marketplace portal for example). This way we at least keep increasing coverage in the future.

File: src/__mocks__/emptyModule.tsx [hunk 3]
```diff
@@ -0,0 +1,2 @@
+// Copyright (c) 2018 Ultimaker B.V.
+module.exports = {}
```
Comment (ChrisTerBeke) [inline src/__mocks__/emptyModule.tsx]: In some places we use ;, in some places we don't. Let's settle for 1 style at least within this library.

File: src/components/__tests__/button.test.tsx [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+    wrapper.setProps({ type: 'submit' });
+    expect(wrapper.find('[type="submit"]')).toHaveLength(1);
+  });
+
+  it('renders a anchor when prop `type` is link', () => {
+    wrapper.setProps({ type: 'link', linkURL: 'https://ultimaker.com/' });
+    expect(wrapper).toMatchSnapshot();
+  });
+
+  it('is disabled when prop `disabled` is true', () => {
```
Comment (ChrisTerBeke) [inline src/components/__tests__/button.test.tsx]: Maybe it's worth testing that it does not call the onClickHandler when the button is disabled?

File: src/stories/icons.stories.tsx [hunk 5]
```diff
@@ -15,14 +15,6 @@ import PendingIcon from '../components/icons/pending_icon';
 import RejectedIcon from '../components/icons/rejected_icon';
 import WarningIcon from '../components/icons/warning_icon';
 import WebsiteIcon from '../components/icons/website_icon';
-import DownloadIcon from '../components/icons/download_icon';
```
Comment (ChrisTerBeke) [inline src/stories/icons.stories.tsx]: Why were all the icons removed?

**Pull request closed without merging**